presto-main/src/main/java/com/facebook/presto/operator/DynamicFilterSourceOperator.java

```diff
@@ -197,6 +197,7 @@ private DynamicFilterSourceOperator(
             this.blockBuilders[channelIndex] = type.createBlockBuilder(null, EXPECTED_BLOCK_BUILDER_SIZE);
             this.valueSets[channelIndex] = new TypedSet(
                     type,
+                    Optional.empty(),
```

Comment:
Do we ever collect dynamic filters for complex types (e.g.: rows?)

Reply:
I'm not sure, but dynamic filter needs to match join semantics, and joins fail on indeterminate arrays/rows, so dynamic filter should do the same.

examples:
```
presto> with input as (SELECT * FROM (VALUES (ARRAY[null], 2), (ARRAY[null], 3))t(x, y)) SELECT * from input i1 join input i2 on i1.x = i2.x;
                    

Query 20240614_131341_75681_9w2fv failed: ARRAY comparison not supported for arrays with null elements

presto> with input as (SELECT * FROM (VALUES (ROW(null), 2), (ROW(null), 3))t(x, y)) SELECT * from input i1 join input i2 on i1.x = i2.x;

Query 20240614_131414_75748_9w2fv failed: ROW comparison not supported for fields with null elements
```